### PR TITLE
Set network pre-process 

### PR DESCRIPTION
**Pull request by gsvr**
To allow for altering through the networks if ported

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: recharges/tasks.py [hunk 1]
```diff
@@ -206,37 +240,25 @@ def run(self, recharge_id, **kwargs):
...
-                                                       countdown=5*60)
-                    return "Recharge for %s: Queued at Hotsocket "\
-                        "#%s" % (recharge.msisdn, hotsocket_ref)
+            l.info("Making hotsocket recharge request")
+            result = self.request_hotsocket_recharge(recharge_id)
+
+            if "hotsocket_ref" not in result["response"]:
+                l.info("Hotsocket error: %s" %
+                       result["response"]["message"])
+                # todo test this
```
Comment (gsvr) [inline recharges/tasks.py]: Who put this here?

Comment (adrianmoisey) [inline recharges/tasks.py]: Looks like it was you!

Comment (gsvr) [inline recharges/tasks.py]: Shh, don't tell Mike.

File: recharges/tasks.py [hunk 2]
```diff
@@ -56,11 +56,10 @@ def normalize_msisdn(msisdn, country_code='27'):
     return msisdn
 
 
-def look_up_mobile_operator(msisdn):
+def slice_network_operator(msisdn):
```
Comment (imsickofmaps) [inline recharges/tasks.py]: It's not a slice! Change it. :palm_tree:

File: recharges/tasks.py [hunk 3]
```diff
@@ -92,10 +91,45 @@ def look_up_mobile_operator(msisdn):
...
+
+        # Normalize the msisdn
+        recharge.msisdn = normalize_msisdn(recharge.msisdn, '27')
+        recharge.save()
+
+        # Set the network operator
+        network = slice_network_operator(recharge.msisdn)
+        if not network:
+            # If no network is found, mark the recharge unrecoverable
+            # gsvr: Perhaps we should rather use a different status here
```
Comment (imsickofmaps) [inline recharges/tasks.py]: We'll add a status field that we can dump reasons for status ID in - we need that for Hotsocket response parsing too

**Comment by imsickofmaps** [comment]
Minor rename then :+1:

**Pull request merged**